packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,7 +178,7 @@ class RequestWrapper {
    */
   async fetch({request}) {
     assert.atLeastOne({request});
-
+    const clonedRequest = request.clone();
```

Comment:
Can you check `this.pluginCallbacks.fetchDidFail` here and only clone the request if there's one or more callbacks? Something like:

```js
const clonedRequest = this.pluginCallbacks.fetchDidFail ? request.clone() : null;
```